panther_lights/src/driver_node.py

```diff
@@ -0,0 +1,179 @@
+
+        # setup and activate panels
+        self._setup_panels()
+
+        # clear panels
+        self._clear_panel(LEDConstants.PANEL_FRONT)
+        self._clear_panel(LEDConstants.PANEL_REAR)
+
+        # -------------------------------
+        #   Subscribers
+        # -------------------------------
+
+        self._front_frame_sub = rospy.Subscriber(
+            'lights/driver/front_panel_frame', Image, self._front_frame_cb, queue_size=1
+        )
+        self._rear_frame_sub = rospy.Subscriber(
+            'lights/driver/rear_panel_frame', Image, self._rear_frame_cb, queue_size=1
+        )
+
+        # -------------------------------
+        #   Services
+        # -------------------------------
+
+        self._set_brightness_service = rospy.Service(
+            'lights/driver/set/brightness', SetLEDBrightness, self._set_brightness_cb
+        )
+
+        rospy.loginfo(f'[{rospy.get_name()}] Node started')
+
+    def _percent_to_apa_driver_brightness(self, brightness_percent: float) -> int:
+        if 0 <= brightness_percent <= 1:
+            brightness = int(brightness_percent * LEDConstants.LED_MAX_BRIGHTNESS)
+            # set minimal brightness for small values
+            if brightness == 0 and brightness_percent > 0:
+                brightness = 1
+            return brightness
+        raise ValueError('Brightness out of range <0,1>')
+
+    def _setup_panels(self) -> None:
+        GPIO.setmode(GPIO.BCM)
+        GPIO.setup(LEDConstants.LED_SWITCH_PIN, GPIO.OUT)
+        GPIO.setup(LEDConstants.LED_POWER_PIN, GPIO.OUT)
+        GPIO.output(
+            (LEDConstants.LED_SWITCH_PIN, LEDConstants.LED_POWER_PIN),
+            (
+                LEDConstants.LED_SWITCH_FRONT_STATE,
+                LEDConstants.LED_POWER_ON_STATE,
+            ),
+        )
+
+    def _front_frame_cb(self, image: Image) -> None:
+        if (rospy.Time.now() - image.header.stamp) > rospy.Duration(self._frame_timeout):
+            rospy.logwarn(f'[{rospy.get_name()}] Front frame timeout exceeded, ignoring frame.')
+            return
+        rgb_frame, brightness = self._decode_img_msg(image)
+        self._set_panel_frame(LEDConstants.PANEL_FRONT, rgb_frame, brightness)
+
+    def _rear_frame_cb(self, image: Image) -> None:
+        if (rospy.Time.now() - image.header.stamp) > rospy.Duration(self._frame_timeout):
+            rospy.logwarn(f'[{rospy.get_name()}] Rear frame timeout exceeded, ignoring frame.')
+            return
+        rgb_frame, brightness = self._decode_img_msg(image)
+        self._set_panel_frame(LEDConstants.PANEL_REAR, rgb_frame, brightness)
+
+    def _decode_img_msg(self, image: Image) -> tuple:
+        rgb_frame = [[image.data[i + j] for j in range(3)] for i in range(0, len(image.data), 4)]
+        brightness = int(image.data[3] / 255 * 100)
+        return rgb_frame, brightness
+
+    def _set_panel_frame(self, panel_num: int, panel_frame: list, brightness: int = 100) -> None:
```

Comment:
```suggestion
    def _set_panel_frame(self, panel_num: int, panel_frame: list, brightness: int = 255) -> None:
```

REscaling the value from 0-255 to 0-100 I would do in this function